test/sanity/validate-modules/ignore.txt

```diff
@@ -435,6 +435,7 @@ lib/ansible/modules/cloud/cloudstack/cs_storage_pool.py E326
 lib/ansible/modules/cloud/cloudstack/cs_template.py E324
 lib/ansible/modules/cloud/cloudstack/cs_template.py E325
 lib/ansible/modules/cloud/cloudstack/cs_template.py E326
+lib/ansible/modules/cloud/cloudstack/cs_template_facts.py E324
```

Comment:
An ignore shouldn't be added for this. The underlying module_utils code should be corrected:

https://github.com/ansible/ansible/blob/5d0ed38e3ad3c4b83bb0c083a8832f8a926da31c/lib/ansible/module_utils/cloudstack.py#L37-L45

I wasn't sure what to do with this at first, since the defaults are dynamic. However, after thinking about it more, I think this is what needs to be done (in a separate PR):

1. Make the defaults static.
2. Update the docs fragment to reflect those values.
3. Update the code to override defaults using env vars and config to maintain backwards compatibility.
4. Remove all the existing ignores that will now be obsolete since the docs match the defaults.

Reply:
I wasn't sure either. I thought about this for a while and I came to the conclusion I can not make the default static without changing the precedence.

I am gonna remove the static default in the docs and to reflect that we have more like fall back value rather then a default one.